crates/ruma-events/src/media_preview_config.rs

```diff
@@ -0,0 +1,168 @@
+//! Types for the [`m.media_preview_config`] event.
+//!
+//! [`m.media_preview_config`]: https://github.com/matrix-org/matrix-spec-proposals/pull/4278
+
+use ruma_macros::StringEnum;
+use serde::{Deserialize, Serialize};
+
+use crate::{macros::EventContent, PrivOwnedStr};
+
+/// The content of an `m.media_preview_config` event.
+#[cfg(feature = "unstable-msc4278")]
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "m.media_preview_config", kind = GlobalAccountData)]
+pub struct MediaPreviewConfigEventContent {
+    /// The media previews configuration.
+    pub media_previews: MediaPreviews,
```

Comment:
We still need to set `#[serde(default)]` on the fields for serde to use the default value of the enum if the field is missing.